sharedbuffers/mapped_struct.py

```diff
@@ -1080,19 +1080,18 @@ def __len__(self):
             return self.bitlen
 
     @cython.ccall
-    @cython.locals(dcode=cython.char, offset=cython.size_t)
-    def _contains(self, elem, dcode, offset):
-        xlen = len(self.objlist)
-        hint = xlen // 2
+    @cython.locals(dcode=cython.char, offset=cython.size_t,
+        start=cython.size_t, xlen=cython.size_t, hint=cython.size_t)
+    def _search_key(self, elem, dcode, offset, start, xlen, hint):
         pindex = cython.cast(cython.p_char, self.objlist.pybuf.buf) + offset
         if dcode == 'q':
-            return _c_search_hkey_i64(elem, pindex, 8, xlen, hint)
+            return _c_search_hkey_i64(elem, pindex + start * 8, 8, xlen, hint)
```

Comment:
`_c_search_hkey_X` methods only exist in cythonized code. This method looks like it shouldn't be called from pure-python ever, so you can make it `cfunc` instead of `ccall`. The calling convention will be faster, and it won't be visible from python, so people cannot shoot themselves in the foot.

Reply:
Good idea.